### PR TITLE
add method to get progressive log

### PR DESCRIPTION
**Pull request by zz-bmj**
With this two method, we can get the progressive log with a constant request.

### CODE REVIEW

**Comment by zz-bmj** [comment]
@supereagle @bndr PTLA

File: build.go [hunk 1]
```diff
@@ -212,6 +212,40 @@ func (b *Build) GetConsoleOutput() string {
 	return content
 }
 
+// GetProgressiveText gets progressive text with the param in previous response header.
```
Comment (supereagle) [inline build.go]: What is `previous response header`? The comment is confused, please make it clearer.

File: build.go [hunk 2]
```diff
@@ -212,6 +212,40 @@ func (b *Build) GetConsoleOutput() string {
...
+func (b *Build) GetProgressiveText(start int64) (string, string, error) {
+	startStr := strconv.FormatInt(start, 10)
+	url := b.Base + "/logText/progressiveText"
+
+	var content string
+	querymap := make(map[string]string)
```
Comment (supereagle) [inline build.go]: s/querymap/queryMap

File: build.go [hunk 3]
```diff
@@ -212,6 +212,40 @@ func (b *Build) GetConsoleOutput() string {
...
+	querymap["start"] = startStr
+
+	response, err := b.Jenkins.Requester.Get(url, &content, querymap)
+	if err != nil {
+		return "", "", err
+	}
+	textSize := response.Header.Get("X-Text-Size")
```
Comment (supereagle) [inline build.go]: What is `textSize` for? Which case need it?

File: build.go [hunk 4]
```diff
@@ -212,6 +212,40 @@ func (b *Build) GetConsoleOutput() string {
...
+	return content, textSize, nil
+}
+
+// GetProgressiveHTML gets progressive html with the param in previous response header.
+func (b *Build) GetProgressiveHTML(start int64) (string, string, error) {
+	startStr := strconv.FormatInt(start, 10)
+	url := b.Base + "/logText/progressiveHtml"
+
+	var content string
```
Comment (supereagle) [inline build.go]: The common code between `GetProgressiveText()` and `GetProgressiveHTML()` can be put in a function.

File: build.go (hunk 3 above)
Comment (zz-bmj) [reply build.go]: The textSize is the length of the log we get, we can use it as next start so that we can get progressive log without repeat.

**Pull request closed without merging**